NebulaPatcher/Patches/Dynamic/FactorySystem_Patch.cs

```diff
@@ -0,0 +1,31 @@
+﻿using HarmonyLib;
+using NebulaPatcher.Patches.Transpilers;
+using NebulaWorld;
+
+namespace NebulaPatcher.Patches.Dynamic
+{
+    [HarmonyPatch(typeof(FactorySystem))]
+    class FactorySystem_Patch
+    {
+        [HarmonyPrefix]
+        [HarmonyPatch(nameof(FactorySystem.RemoveInserterComponent))]
+        public static bool RemoveInserterComponent_Prefix(FactorySystem __instance, int id)
+        {
+            if (!SimulatedWorld.Initialized || !LocalPlayer.IsMasterClient)
+            {
+                return true;
+            }
+
+            foreach (var obj in InserterComponent_Transpiler.FaultySortersText)
+            {
+                if (obj.Value?.transform.position == __instance.inserterPool[id].pos2)
+                {
```

Comment:
Don't use null conditional with unity types. It breaks at runtime since they override ==